iothub/device/src/Edge/DeviceClientFactory.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            this.transportType = transporType;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeviceClientFactory"/> class with transport settings.
+        /// </summary>
+        /// <param name="transportSettings">Prioritized list of transportTypes and their settings.</param>
+        public DeviceClientFactory(ITransportSettings[] transportSettings)
+        {
+            this.transportSettings = transportSettings;
+        }
+
+        /// <summary>
+        /// Creates a DeviceClient instance based on environment.
+        /// </summary>
+        /// <returns></returns>
+        public DeviceClient Create()
+        {
+            return this.CreateDeviceClientFromEnvironment();
+        }
+
+        DeviceClient CreateDeviceClientFromEnvironment()
+        {
+            IDictionary envVariables = Environment.GetEnvironmentVariables();
+
+            string connectionString = this.GetValueFromEnvironment(envVariables, EdgehubConnectionstringVariableName) ?? this.GetValueFromEnvironment(envVariables, IothubConnectionstringVariableName);
+
+            // First try to create from connection string and if env variable for connection string is not found try to create from edgedUri
+            if (!string.IsNullOrWhiteSpace(connectionString))
+            {
+                return this.CreateDeviceClientFromConnectionString(connectionString);
+            }
+            else
+            {
+                string edgedUri = this.GetValueFromEnvironment(envVariables, IotEdgedUriVariableName) ?? throw new InvalidOperationException($"Environement variable {IotEdgedUriVariableName} is required.");
+                string deviceId = this.GetValueFromEnvironment(envVariables, DeviceIdVariableName) ?? throw new InvalidOperationException($"Environement variable {DeviceIdVariableName} is required.");
+                string moduleId = this.GetValueFromEnvironment(envVariables, ModuleIdVariableName) ?? throw new InvalidOperationException($"Environement variable {ModuleIdVariableName} is required.");
+                string hostname = this.GetValueFromEnvironment(envVariables, IotHubHostnameVariableName) ?? throw new InvalidOperationException($"Environement variable {IotHubHostnameVariableName} is required.");
+                string authScheme = this.GetValueFromEnvironment(envVariables, AuthSchemeVariableName) ?? throw new InvalidOperationException($"Environement variable {AuthSchemeVariableName} is required.");
+                string gateway = this.GetValueFromEnvironment(envVariables, GatewayHostnameVariableName);
+                string apiVersion = this.GetValueFromEnvironment(envVariables, IotEdgedApiVersionVariableName);
+
+                if (!string.Equals(authScheme, SasTokenAuthScheme, StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new InvalidOperationException($"Unsupported authentication scheme. Supported scheme is {SasTokenAuthScheme}.");
```

Comment:
SAS token is the only supported auth scheme for now when using edged. If connection string is set in the env variable it will use that to connect to Edge Hub/IoT Hub

Reply:
Yes, SAS token is the only auth scheme supported by EdgeLet now.. and that is what this Factory is really giving us.